clang/runtime/dpct-rt/include/dpct/blas_gemm_utils.hpp

```diff
@@ -222,6 +229,31 @@ class matmul_desc_t {
 
 namespace detail {
 /// Sacling each row of matrix D with the corresponding element of vector alpha.
+
+template <typename primitive_type, typename... args_type>
+inline
+typename primitive_type::primitive_desc sync_gelu_backward(
+    float alpha, float beta, const::dnnl::memory_desc_ext &src_desc,
+    const ::dnnl::memory_desc_ext &dest_desc) {
+
+  auto alg = ::dnnl::algorithm::eltwise_gelu_erf;
+  return create_primitive_desc<primitive_type>(
+                    ::dnnl::prop_kind::backward, alg, src_desc.get_desc(),
+                    dst_desc.get_desc(), alpha, beta);
+}
+
+template <typename primitive_type, typename... args_type>
+inline
+typename primitive_type::primitive_desc bias_backward(
+    float alpha, float beta, const dnnl::memory_desc_ext &src_desc,
+    const ::dnnl::memory_desc_ext &dest_desc) {
+
+  auto alg = ::dnnl::algorithm::reduction_sum;
+  return create_primitive_desc<primitive_type>(
+                    ::dnnl::prop_kind::backward, alg, src_desc.get_desc(),
+                    dst_desc.get_desc(), alpha, beta);
+}
```

Comment:
For backward primitives, it needs a forward primitive object as a constructor argument.